src/components/NewService.vue

```diff
@@ -122,14 +122,15 @@
       </p>
       <span class="select">
         <select v-model="configuration[linkedService.as]">
-          <option v-for="service in services" v-bind:value="service[linkedService.retrieve]">
+          <option v-for="service in filterServicesByType(linkedService.service_type)" v-bind:value="service[linkedService.retrieve]">
             {{ service.name }}
           </option>
         </select>
       </span>
-      <p class="text-danger" v-if="services.length === 0">You must register some services to link to this one</p>
       <hr/>
     </div>
+    <p class="text-danger" v-if="services.length === 0 || noLinkableServices">You don't have the required dependent Services registered in order to deploy this Service</p>
+    <hr v-if="services.length === 0 || noLinkableServices"/>
     <button class="button is-primary" v-show="!deploying" v-on:click="submit" :disabled="errors.any() || cannotRegister || noLinkableServices">Register Service</button>
```

Comment:
This part is a bit tricky  to read. Some questions:
- "Disabled condition" of Register and Register and Deploy look the same. Is errors being filled differently?
- The error message might not be clarifying the overall workflow conditions to the user. Let's see...
  - To deploy the service X you actually need all the services it depends on being Deployed (not just Registered), right?
  - Can user register a service even if it's not yet able to deploy it for some reasons? I used to think yes.
  - Deployed implies registered, not the other way around, right?.

Is there maybe a way to factor out the logic condition checks in a function (to avoid having it scattered in the disabled/enabled parts of different components).

Reply:
@taliaga 

- That part is to disable the buttons to register (or register and deploy) when there are not enough required services registered. If you can't provide an Orion instance when required you can't register an IoT Agent for example.
- A service can be registered but not deployed. Right now I haven't added the check for actual deployment yet, just if they are registered. Checking for actual deployment needs a bit more work, so I'll do it in a later pull request.

And with Vue.js I found it more convenient in the long run to have disabled/enabled as part of the components, because the rendering of the components will always call to check for these values.

Reply:
> That part is to disable the buttons to register (or register and deploy) when there are not enough required services registered. If you can't provide an Orion instance when required you can't register an IoT Agent for example.

Thanks for the explanation!. OK so without a pointer to an Orion instance not even "register" is possible...makes sense.

> A service can be registered but not deployed. Right now I haven't added the check for actual deployment yet, just if they are registered. Checking for actual deployment needs a bit more work, so I'll do it in a later pull request.

I see. I believe you could somehow leverage on the healthcheck script for this, let's see.

> And with Vue.js I found it more convenient in the long run to have disabled/enabled as part of the components, because the rendering of the components will always call to check for these values.

Sure, I was not worried about having disabled/enabled in particular, it was more a concern with reducing the scattering/duplication of logic checks. For example, could you have both enabled and disabled sharing the result of a function call? things like this.